dummyfs/dummyfs.c

```diff
@@ -337,12 +337,13 @@ int dummyfs_link(void *ctx, oid_t *dir, const char *name, oid_t *oid)
 	}
 
 	if (ret != EOK) {
-		object_unlock(fs, d);
+		object_unlock(fs, d); /* FIXME: remove after adding per-object locking */
```

Comment:
Please note that `object_unlock(fs, d)` call here is just so that we could lock `o` object in the very next line.